### PR TITLE
N21-1774 Create synced course

### PR DESCRIPTION
**Pull request by MarvinOehlerkingCap**
# Short Description

## Links to Ticket and related Pull-Requests
- https://ticketsystem.dbildungscloud.de/browse/N21-1774
- https://github.com/hpi-schul-cloud/schulcloud-client/pull/3426
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/817
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/4830

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/data-group/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@ export * from "./types";
 export * from "./GroupState.composable";
 export * from "./GroupApi.composable";
 export * from "./GroupMapper";
+export * from "./groupListState.composable";
```
Comment (arnegns) [inline src/components/data-group/index.ts:5]: Is the naming of the composable correct?

File: src/components/data-group/types/paginationOptions.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface PaginationOptions {
```
Comment (arnegns) [inline src/components/data-group/types/paginationOptions.ts:1]: Looks like a shared interface. Do we have a place for that in the new folder structure?

File: src/components/feature-course-sync/GroupSelectionDialog.vue [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+			<div class="text-h4 my-2 text-break">
+				{{ $t("feature-course-sync.GroupSelectionDialog.title") }}
+			</div>
+		</template>
+
+		<template #content>
+			<p class="text-md mt-2">
+				{{ $t("feature-course-sync.GroupSelectionDialog.text") }}
+			</p>
+			<VAutocomplete
```
Comment (arnegns) [inline src/components/feature-course-sync/GroupSelectionDialog.vue:20]: Do we need an aria label here?

File: src/components/templates/RoomWrapper.vue [hunk 4]
```diff
@@ -29,99 +29,113 @@
...
 		</template>
-		<common-cartridge-import-modal :max-width="480" class="upload-modal" />
-	</default-wireframe>
+		<GroupSelectionDialog v-model:is-open="isCourseSyncDialogOpen" />
+		<CommonCartridgeImportModal :max-width="480" class="upload-modal" />
+	</DefaultWireframe>
 </template>
 
-<script>
+<script setup lang="ts">
```
Comment (arnegns) [inline src/components/templates/RoomWrapper.vue:37]: Nice :) !

**Review by arnegns — APPROVED** [review]

File: src/components/data-group/index.ts (hunk 1 above)
Comment (MarvinOehlerkingCap) [reply src/components/data-group/index.ts:5]: yes we use camelCase now

File: src/components/feature-course-sync/GroupSelectionDialog.vue (hunk 3 above)
Comment (MarvinOehlerkingCap) [reply src/components/feature-course-sync/GroupSelectionDialog.vue:20]: I believe that vuetify components already have a11y in mind and I would assume that it uses the label prop

**Review by mrikallab — APPROVED** [review]

**Pull request merged**